### DISCUSSION TITLE
Should renovate recognize pnpm at first glance?

### DISCUSSION THREAD

**Discussion by rgrzesk** [renovatebot/renovate#21262] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I am trying to integrate renovate with my project based on pnpm. The Renovate is installed on github, project runs with pnpm, but the first PR always says that it is npm related and within detected package files recognizes only:
```
.github/workflows/chromatic.yml (github-actions)
package.json (npm)
.nvmrc (nvm)
```
Repository contains also `pnpm-lock.yaml`. Should that be included as well in the first PR?

### What you tried so far.

I have added to package.json:
```
"engines": {
        "node": ">=16",
        "pnpm": ">=7"
    }
```
and 
```
"pnpm": {
        "peerDependencyRules": {
            "ignoreMissing": [
                "somePackage"
            ]
        }
    }
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
renovate has a single `npm` manager for `package.json`. that manager supports `npm`, `pnpm` and `yarn`. it determine via lockfile.

**Reply by rgrzesk** [reply]
does that mean that I could merge the PR and later on pnpm-lock file should be recognized?

**Reply by rgrzesk** [reply]
@viceice once having `pnpm-lock.yaml` in the repository I guess I should see `package.json (pnpm)` not `package.json (npm)` as I see now, correct? Or that report is generic and in the second run it will recongize correctly?

**Reply by viceice** [reply]
it's only report inside debug logs. renovate will manually update the package. then if a lockfile is found, renovate will call the appropriate package manager to update the lockfile.

the renovate manager still names npm and don't change.

**Reply by rgrzesk** [reply]
Cool, thank you :)
